script.js

```diff
@@ -1,4 +1,495 @@
+  var suits = ["hearts", "diamonds", "clubs", "spades"];
+  var emojiLibrary = ["♥", "♦", "♣", "♠"];
+  // Loop over the suits array
+  var suitIndex = 0;
+  while (suitIndex < suits.length) {
+    // Store the current suit in a variable
+    var currentSuit = suits[suitIndex];
+    var currentEmoji = emojiLibrary[suitIndex];
+    // Loop from 1 to 13 to create all cards for a given suit
+    // Notice rankCounter starts at 1 and not 0, and ends at 13 and not 12.
+    // This is an example of a loop without an array.
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      // By default, the card name is the same as rankCounter
+      var cardName = rankCounter;
+      var cardValue = rankCounter;
+      // If rank is 1, 11, 12, or 13, set cardName to the ace or face card's name
+      if (cardName == 1) {
+        cardName = "Ace";
+      } else if (cardName == 11) {
+        cardName = "Jack";
+        cardValue = 10;
+      } else if (cardName == 12) {
+        cardName = "Queen";
+        cardValue = 10;
+      } else if (cardName == 13) {
+        cardName = "King";
+        cardValue = 10;
+      }
+      // Create a new card with the current name, suit, and rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        value: cardValue,
+        emoji: currentEmoji,
+      };
+      // Add the new card to the deck
+      cardDeck.push(card);
+      // Increment rankCounter to iterate over the next rank
+      rankCounter += 1;
+    }
+    // Increment the suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+  // Return the completed card deck
+  return cardDeck;
+};
+
+//shuffling deck
+var shuffleDeck = function (cardDeck) {
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    //switch the 2 cards around by swapping like this:
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+
+    currentIndex += 1;
+  }
+  return cardDeck;
+};
+
+// helper function: for shuffling deck
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+//helper function: reset the game for new game loop
+var resetGame = function () {
+  gameMode = INTRO_MODE;
+  playerHand = [];
+  computerHand = [];
+};
+
+// check bust or blackjack in any hand
+var checkBustOrBlackjack = function (playerHand, computerHand) {
+  var deckValue1 = addAllCards(playerHand);
+  var deckValue2 = addAllCards(computerHand);
+
+  //see if any hand is 21 or more
+
+  if (deckValue1 >= 21 || deckValue2 >= 21) {
+    //gameMode = RESULT_MODE;
+    gameMode = RESULT_MODE;
+  }
+
+  console.log("game mode now", gameMode);
+  return gameMode;
+};
```

Comment:
Seems like this function is not called in your main function (can remove if unnecessary)! Also, returning the `gameMode` here seems weird since you are already changing the gameMode in line 189. If there's nothing to return, the function can don't return anything as well.